### PR TITLE
Container weights: https://blueprints.launchpad.net/or/+spec/container-weights

### PR DESCRIPTION
**Pull request by Csantucci**
See also http://www.elvastower.com/forums/index.php?/topic/36162-container-management/page__view__findpost__p__291153

### CODE REVIEW

File: Source/Documentation/Manual/features-rollingstock.rst
Comment (cesarBLG) [inline Source/Documentation/Manual/features-rollingstock.rst:640]: How is this different from the Mass parameter defined in the Wagon section of the .wag file?

File: Source/Orts.Simulation/Simulation/Container.cs
Comment (cesarBLG) [inline Source/Orts.Simulation/Simulation/Container.cs:554]: Why don't you just perform a linear mapping?
```csharp
container.MassKG = container.EmptyMassKG + (container.MaxMassWhenLoadedKG-container.EmptyMassKG)*loadPercent/100f;
```

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs
Comment (cesarBLG) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs:861]: Probably it would be cleaner if you use LINQ. However, this is a personal preference, so it's up to you.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs
Comment (cesarBLG) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs:1983]: Consider whether this code duplication can be removed by creating a CalculateMass() function. Same code here as in LoadFromWagFile.

File: Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/FreightAnimations.cs
Comment (cesarBLG) [inline Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/FreightAnimations.cs:326]: Same comment as before regarding the linear map. Please consider also eliminating the code duplication.

**Review by cesarBLG — CHANGES_REQUESTED** [review]
Thanks Carlo,
I leave you a few comments here.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs
Comment (Csantucci) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs:861]: I have few knowledges about LINQ, so I prefer to leave it so.

File: Source/Documentation/Manual/features-rollingstock.rst
Comment (Csantucci) [reply Source/Documentation/Manual/features-rollingstock.rst:640]: Mass () in .wag file is an original MSTS parameter. It may or may not include freight weight, depending from what the author of the wagon decided. So it is not a reliable value for the empty wagon weight.
Moreover WagonEmptyWeight has been introduced already in the original FreightAnimations code; so it is already used and it is not possible to remove it now.

File: Source/Orts.Simulation/Simulation/Container.cs
Comment (Csantucci) [reply Source/Orts.Simulation/Simulation/Container.cs:554]: Sorry, but I don't understand your point. I assume we are speaking about the LoadState.Random case.
With my code there is 30% of probability that the container is considered as empty, while there is a 70% of probability that the container weighs between 30% and 100% of the max allowable weight, because I assume that in general the weight of the freight is at least 30% of the weight of the max allowable weight. These are of course only assumptions, but I think they can be accepted.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs
Comment (Csantucci) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs:1989]: Code duplication removed.

File: Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/FreightAnimations.cs
Comment (Csantucci) [reply Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/FreightAnimations.cs:313]: Code duplication removed.

**Comment by Csantucci** [comment]
> Thanks Carlo, I leave you a few comments here.

Thanks for the review. Please see my replies.

File: Source/Documentation/Manual/features-rollingstock.rst
Comment (cesarBLG) [reply Source/Documentation/Manual/features-rollingstock.rst:640]: Thanks for the explanation

File: Source/Orts.Simulation/Simulation/Container.cs
Comment (cesarBLG) [reply Source/Orts.Simulation/Simulation/Container.cs]: Ah, I see, thanks.

**Review by cesarBLG — APPROVED** [review]
This is a nice addition! Thanks also for your clarifications. Waiting for blueprint approval.

**Comment by Csantucci** [comment]
> This is a nice addition! Thanks also for your clarifications. Waiting for blueprint approval.

Thanks for your time! Let's wait for the blueprint approval.

**Pull request merged**
